src/viperleed/gui/measure/widgets/measurement_plot.py

```diff
@@ -255,35 +265,35 @@ def __plot_all_time_resolved_data(self):
         if not has_data:
             return
 
-        self.__ctrl_color = dict(zip(data, _COLORS))
+        self._store_markers_for_devices(data.keys())
 
-        for marker, quantity in zip(self.__markers, self.plotted_quantities):
+        for color, quantity in zip(_COLORS, self.plotted_quantities):
             for ctrl, measurements in data.items():
                 if quantity not in measurements:
                     continue
                 ctrl_data = measurements[quantity]
                 ctrl_times = measurements[QuantityInfo.TIMES]
 
                 if not SEPARATE_STEPS:
-                    color = self.__ctrl_color[ctrl](COLOR_FRACTION)
-                    style = _marker_style(*marker, color)
+                    plot_color = color(COLOR_FRACTION)
+                    style = self._get_marker_style_for_device(ctrl, plot_color)
                     self.lines[quantity][ctrl], = axes.plot(ctrl_times,
                                                             ctrl_data, **style)
                     continue
 
                 # SEPARATE_STEPS
-                colors = self.__ctrl_color[ctrl](
-                    np.linspace(0.2, 0.8, len(ctrl_data))
+                colors = color(
+                    np.linspace(0.2, 0.8, self.data_points.nr_steps_total)
                     )
                 # Cannot construct a big array of arrays and plot in
-                # parallel bacause each step may (and usually will)
+                # parallel because each step may (and usually will)
                 # have a variable number of data.
-                for color, times, values in zip(colors, ctrl_times, ctrl_data):
-                    style = _marker_style(*marker, color)
+                for clr, times, values in zip(colors, ctrl_times, ctrl_data):
+                    style = self._get_marker_style_for_device(ctrl, clr)
```

Comment:
```suggestion
                for step_color, times, values in zip(colors, ctrl_times, ctrl_data):
                    style = self._get_marker_style_for_device(ctrl, step_color)
```

`clr` looks like a typo for `ctrl` that we use in many other spots